src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/AddressDtoValidator.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.companieshouse.overseasentitiesapi.validation;
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.Country;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.StringValidators;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+@Component
+public class AddressDtoValidator {
+
+    public Errors validate(String parentAddressField, AddressDto addressDto, Errors errors, String loggingContext) {
+        validatePropertyNameNumber(parentAddressField, addressDto.getPropertyNameNumber(), errors, loggingContext);
+        validateLine1(parentAddressField, addressDto.getLine1(), errors, loggingContext);
+        validateLine2(parentAddressField, addressDto.getLine2(), errors, loggingContext);
```

Comment:
Before validating `line2` we need to make sure that the value is not null to avoid exception, and we should not use the `validateStringNotBlank` because is not a mandatory field

Reply:
Do we need this for the other optional fields